Source/WebCore/page/WorkerNavigator.cpp

```diff
@@ -66,6 +68,15 @@ void WorkerNavigator::setAppBadge(std::optional<unsigned long long> badge, Ref<D
         return;
     }
 
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+    if (is<ServiceWorkerGlobalScope>(*scope)) {
+        if (auto pushNotificationEvent = RefPtr { downcast<ServiceWorkerGlobalScope>(*scope).pushNotificationEvent() }) {
```

Comment:
nit: little bit of a weird pattern. This would be more concise and achieve the same:
```
if (RefPtr event = downcast<ServiceWorkerGlobalScope>(*scope).pushNotificationEvent()) {
```